x/ccv/provider/keeper/keeper.go

```diff
@@ -249,9 +249,16 @@ func (k Keeper) GetAllConsumerChains(ctx sdk.Context) (chains []types.Chain) {
 		chainID := string(iterator.Key()[1:])
 		clientID := string(iterator.Value())
 
+		// Get the consumer TopN value by checking
+		// if the chain is a TopN chain,
+		// otherwise it's a OptIn chain
+		topN, isTopN := k.GetTopN(ctx, chainID)
+
 		chains = append(chains, types.Chain{
 			ChainId:  chainID,
 			ClientId: clientID,
+			Opt_In:   !isTopN,
```

Comment:
I think this should be:
```suggestion
			Opt_In:   !isTopN || topN == 0,
```
`GetTopN` would return a `found` on whether `TopN` is set, but `TopN` could be set to `0`.

Reply:
Similarly, below in `keeper_test.go` you probably won't need to check `if topN > 0 {` in

```
if topN > 0 {
    pk.SetTopN(ctx, chainID, topN)
```

Reply:
Make sense, thanks. Updated.